src/__tests__/index.test.js

```diff
@@ -69,6 +69,48 @@ describe('duplicate-transitive-replacement', () => {
         );
     });
 
+    it('should not mistakenly replace prefix string', () => {
```

Comment:
This is great, thanks for adding some test coverage for the change. I made a few edits to try and make the code a bit more like a real world example, using hypothetical `radio-button` and `radio-button-group` packages. The test is exactly the same in terms of functionality, what do you think?

```suggestion
    it('should not deduplicate when package name is partial match', () => {
        mockFs({
            [path.resolve(
                'node_modules/@org/component-a/node_modules/@org/radio-button',
                './index.js'
            )]: 'some radio button code',

            [path.resolve(
                'node_modules/@org/component-b/node_modules/@org/radio-button',
                './index.js'
            )]: 'some radio button code',

            [path.resolve(
                'node_modules/@org/component-b/node_modules/@org/radio-button-group',
                './index.js'
            )]: 'some radio button GROUP code',
        });

        const duplicates = [
            // although duplicate packages are prefixed with 'radio-button', these should
            // be ignored as they are not a full match on the 'radio-button-group'
            [
                'node_modules/@org/component-a/node_modules/@org/radio-button',
                'node_modules/@org/component-b/node_modules/@org/radio-button',
            ],
        ];

        const matchingResource = mockResource({
            filename: './index.js',
            context: path.resolve(
                'node_modules/@org/component-b/node_modules/@org/radio-button-group'
            ),
        });

        silent.mockImplementation(() =>
            path.resolve(matchingResource.context, matchingResource.request)
        );

        const finder = require('find-package-json');

        finder.mockImplementation(() => ({
            next: () => ({ value: { name: '@org/component-b' } }),
        }));

        const res = deduplicate(matchingResource, duplicates);

        expect(res).toBeFalsy();
    });
```